### ISSUE TITLE
Support multiple controllers

### ISSUE THREAD

**Issue by rsubtil** [rsubtil/controller_icons#58]
Currently, the addon checks for inputs for the first controller. There should be an option to specify controller index for multiplayer games.

**Comment by mubinulhaque** [comment]
My idea for this:

- Each ControllerIcon node would have an additional variable called joy_index
- There would be an enum called JOYPAD_INDEX that could either be:
   - LAST_JOYPAD_USED (set by default)
   - JOYPAD_0
   - JOYPAD_1
   - JOYPAD_2
   - JOYPAD_3
   - JOYPAD_4
   - JOYPAD_5
   - JOYPAD_6
   - JOYPAD_7
- The user could change it to any of the following for any ControllerIcon node separately

**Comment by TacticalLaptopBag** [comment]
Does this also cover icons disappearing for other controllers that aren't icons for the last controller used? I'm running into issues with my game where this is happening. I'll send a video or picture later tonight to demonstrate.

**Comment by TacticalLaptopBag** [comment]
This is the problem I'm experiencing - I'm using two controllers, one for each character. You can see when one moves, one set of icons show. When the other moves, the other set of icons show. For now, I can resolve this by just setting the textures to their respective Joypad paths, but it's more convenient to have it defined by the action.
[Screencast from 2024-10-12 10-49-21.webm](https://github.com/user-attachments/assets/cd000c40-c88b-40d7-b0bf-830dfcee25e5)

**Comment by rsubtil** [comment]
Hi @TacticalLaptopBag, sorry for the late response. Could you open a dedicated issue describing this problem with more detail (e.g. what controllers are being used, what and how icon paths are being used), ideally with a minimum reproduction project (MRP)?

**Comment by SebMatDo** [comment]
I also had the @TacticalLaptopBag bug, I tried a lot to bypass that, finally I made my custom solution (using this addon as the base and cutting a lot of things).
@rsubtil You are credited :) and sorry for the hard cut of the addon haha.

This is the link of my solution, maybe I help another person https://github.com/SebMatDo/Controllers-Icons-Local-Multiplayer it support 8 different players and while it doesn't remap or change automatically, you can show different icons or inputs for differents players.

**Comment by greenpixels** [comment]
@rsubtil Is this issue resolved with #131 ?

**Comment by rsubtil** [comment]
> [@rsubtil](https://github.com/rsubtil) Is this issue resolved with [#131](https://github.com/rsubtil/controller_icons/issues/131) ?

Indeed, it should be solved now. Thanks for reminding me, and for your work on this!
